packages/utils/src/object.ts

```diff
@@ -63,9 +60,11 @@ export function addNonEnumerableProperty(obj: { [key: string]: unknown }, name:
  * @param original the original function that gets wrapped
  */
 export function markFunctionWrapped(wrapped: WrappedFunction, original: WrappedFunction): void {
-  const proto = original.prototype || {};
-  wrapped.prototype = original.prototype = proto;
-  addNonEnumerableProperty(wrapped, '__sentry_original__', original);
+  try {
+    const proto = original.prototype || {};
+    wrapped.prototype = original.prototype = proto;
+    addNonEnumerableProperty(wrapped, '__sentry_original__', original);
+  } catch (o_O) {} // eslint-disable-line no-empty
```

Comment:
h: I would add a log here (maybe even non-debug!). I can already see myself debugging an issue that some instrumentation isn't working. This is a very opaque way to fail.

Reply:
I can add a debug log, but wouldn't do a log. Because it is really not a user problem, the linked issue:

```js
function callback() {
    console.log("Hello");
}
setTimeout(callback, 50); // OK
Object.freeze(callback);
setTimeout(callback, 50); // BOOM!
```

Is generally totally OK to write but would result in this. the user shouldn't see a log because of this IMHO.

Reply:
Debug log sounds good!

Reply:
I added a debug log & integration tests to actually make sure the issue raised is fixed. Interestingly, if the function if frozen the actual monkey patching still works, only the setting of the wrapped property fails. So the error is still captured even if the callback is frozen, which is good I guess.